cpp/src/arrow/util/value_parsing.h

```diff
@@ -561,6 +562,38 @@ static inline bool ParseTimestampStrptime(const char* buf, size_t length,
   return true;
 }
 
+/// \brief Returns time since the UNIX epoch in the requested unit. Takes null terminated
+/// buffer as argument
+static inline bool ParseTimestampStrptime(const char* buf,
```

Comment:
Line 533 contains the original function that takes non null-terminated string as argument.